### DISCUSSION TITLE
Obsidian.md version

### DISCUSSION THREAD

**Discussion by girdzijauskas** [utk-se/CodeRibbon#95] [General]
I think this project is awesome and has a future.

I'm interested in an Obsidian.md (https://obsidian.md) version of this, as the interface I currently default to in Obsidian is essentially 4 patches. Scrolling either side would be absolutely amazing and super intuitive.

I'm a web dev, though with little to none Electron experience / understand of the internals of either Obsidian or CodeRibbon, but this seems like a good thing to play with, that I also want to see happen. So I'm starting some structural investigation on how to port this over.

If anybody is also interested in seeing this happen, I'm happy to join forces at some level. I've got some experience of React + good enough front-end experience, and am willing to stumble through this even if it takes a long amount of time.

**Comment by robobenklein** [comment]
Obsidian.md looks like an awesome tool, but I doubt I could do much to change it without open access to it's source code. I looked briefly but it doesn't appear to be open source? As such I can't offer much advice about how to implement a ribbon view there.

The furthest I could dig was their API included what looked like accessors to the workspace, so if it is electron-based it should certainly be possible. Since it's a paid application, perhaps a feature request to them could carry more weight?

**Comment by girdzijauskas** [comment]
Indeed, I've posted on the Plugin Ideas part of the community as well, https://forum.obsidian.md/t/coderibbon-plugin/17647

**Comment by robobenklein** [comment]
I have not used *either* of these extensively, but looking at https://github.com/zadam/trilium reminded me of some of the same features I saw valuable from Obsidian, but since it's open it might be easier for me to poke around in to see how hard it'd be to modify into a linear-dimension workspace.

**Reply by girdzijauskas** [reply]
Trillium looks interesting - I'll have to test run it at some point, though from initial setup it looks like it's trying to do a lot of things at once.

I've done my share of Evernoting and trying to come up with organisational systems; the thing that Obsidian does really well I think is the core principles of no proprietary database system - just plain .md files (which mean I can git my work), links being treated as a primary feature, and from there, it's just trying to provide what is essentially a minimal flexible IDE for notes. Which makes it really good for writing and thinking.

**Comment by girdzijauskas** [comment]
I am well on my way - I've got a minimal plugin working locally and currently polishing the hiccups before releasing the plugin. It's not a port of the code here, and more of a 'I really like the ideas of how CodeRibbon works, so let's build them one by one for Obsidian'. 

Speaking of the release - if it isn't a port of your code, but rather my implementation of your ideas, how should I go about crediting you for it? I'm including links as the full inspiration. And naming too - I'm going for something obvious like 'Obsidian CodeRibbon' or the like at the moment. Do you have any input regarding that?

**Reply by robobenklein** [reply]
The "linear space" code editor paradigm is not my original idea, if you'd like to cite that idea as it initially was introduced to me you can look at Dr. Henley's 2014 publication: https://web.eecs.utk.edu/~azh/pubs/Henley2014CHI_Patchworks.pdf

CodeRibbon is essentially a successor / second iteration of that idea, with specific features and usability changes added, like differing number of patches per column and the overview mode. (Notably, PatchWorks Patches were entirely fixed size and could not be freely sized in any way, and there was less functionality for modifying the middle area of Ribbons.)

You're free to use the 'CodeRibbon' name as a descriptor, so long as it implements the same functionality, namely the 'Ribbon' and the linear infinite dimension grid of patches, and remains open source so that we can potentially use it for nonprofit or academic research.

If you'd like to cite this specific implementation we are planning on publishing a paper on the tool / design within the year, and linking to the website will help to increase our userbase and further research efforts!
